common/CurrencyTable/CurrencyTable.tsx

```diff
@@ -0,0 +1,133 @@
+import { Box, Button, ButtonGroup, Input } from "@chakra-ui/react";
+import {
+  Table,
+  TableCaption,
+  TableContainer,
+  Tbody,
+  Td,
+  Tfoot,
+  Th,
+  Thead,
+  Tr,
+} from "@chakra-ui/table";
+import { useEffect, useState } from "react";
+import { getCurrencies } from "./utils";
+
+interface CurrencyTableProps {
+  onRowClick: (currencyCode: string, index: number) => void;
+}
+
+export type Currency = Array<string>;
+
+export type Currencies = Array<Currency>;
+
+export const CurrencyTable = ({ onRowClick }: CurrencyTableProps) => {
+  const [currencies, setCurrencies] = useState<Currencies>([]);
+  const [filteredCurrencies, setFilteredCurrencies] = useState<Currencies>([]);
+  const [pageNumber, setPageNumber] = useState(1);
+  const [paginatedCurrencies, setPaginatedCurrencies] = useState<Currencies>(
+    []
+  );
+  const [searchVal, setSearchVal] = useState<string>("");
```

Comment:
Nit: the `string` type will be inferred from the initial state

```suggestion
  const [searchVal, setSearchVal] = useState("");
```

Reply:
resolved